tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,726 @@
+    mock_api_object.change_output.assert_any_call(output_id, selected=False)
+    mock_api_object.set_volume.assert_any_call(output_id=output_id, volume=30)
+    mock_api_object.set_volume.assert_any_call(output_id=output_id, volume=0)
+    mock_api_object.set_volume.assert_any_call(
+        output_id=output_id, volume=initial_volume
+    )
+    output_id = SAMPLE_OUTPUTS_ON[2]["id"]
+    mock_api_object.change_output.assert_any_call(output_id, selected=True)
+
+
+async def test_last_outputs_master(hass, mock_api_object):
+    """Test restoration of _last_outputs."""
+    # Test turning on sends API call
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TURN_ON)
+    assert mock_api_object.change_output.call_count == 0
+    assert mock_api_object.set_enabled_outputs.call_count == 1
+    await _service_call(
+        hass, TEST_MASTER_ENTITY_NAME, SERVICE_TURN_OFF
+    )  # should have stored last outputs
+    assert mock_api_object.change_output.call_count == 0
+    assert mock_api_object.set_enabled_outputs.call_count == 2
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TURN_ON)
+    assert mock_api_object.change_output.call_count == 3
+    assert mock_api_object.set_enabled_outputs.call_count == 2
+
+
+async def test_bunch_of_stuff_master(hass, mock_api_object, get_request_return_values):
+    """Run bunch of stuff."""
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TURN_ON)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TURN_OFF)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TOGGLE)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_MEDIA_VOLUME_MUTED: True},
+    )
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_MEDIA_VOLUME_MUTED: False},
+    )
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_SET,
+        {ATTR_MEDIA_VOLUME_LEVEL: 0.5},
+    )
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PAUSE)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PLAY)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_STOP)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PREVIOUS_TRACK)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_NEXT_TRACK)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_MEDIA_SEEK,
+        {ATTR_MEDIA_SEEK_POSITION: 35},
+    )
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_CLEAR_PLAYLIST)
+    await _service_call(
+        hass, TEST_MASTER_ENTITY_NAME, SERVICE_SHUFFLE_SET, {ATTR_MEDIA_SHUFFLE: False}
+    )
+    # stop player and run more stuff
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0.2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)